src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -132,7 +132,7 @@ public Command parseCommand(String userInput) throws ParseException {
             return new AddTagCommandParser().parse(arguments);
 
         default:
-            WordSuggestion commandSuggestions = new WordSuggestion(commandWord, COMMAND_WORDS, 3);
+            WordSuggestion commandSuggestions = new WordSuggestion(commandWord, COMMAND_WORDS, 2);
```

Comment:
Very elaborate implementation for WordSuggestion. Perhaps you could explain if changing the distanceLimit from 3 to 2 will fix the WordSuggestion bug? For example, why not change it to 1?

Reply:
It's mainly because of the algorithm I used which is based on the edit distance of the two strings. And [edit distance](https://en.wikipedia.org/wiki/Levenshtein_distance) is basically the minimum number of single-character edits (insertions, deletions or substitutions) required to change one word into the other. And for now I found 2 to be the optimal balance for the edit distance. 

In detail, previously when the max edit distance was 3, the issue was #243 where when a user types in any character, it will always suggest the `add` command, why, because if you notice, the edit distance between any character and `add` is either 2 or 3. For example from `z` to `add`, it has an edit distance of 3, which is first substituting `z` with `a`, second adding `d` and third also adding `d`. Which is why for any single character, it will suggest `add`. The scenario where the edit distance is 2 would be either for `a` or `d`, because it would just be adding two more characters as one character is alrdy correct. Hence I decided to reduce it to two to prevent cases like #243 but also allowing for more flexible suggestions.

Reply:
Sounds good to me. We'll do more testing to confirm that it holds up. Thanks!